src/cml.js

```diff
@@ -159,9 +165,43 @@ class CML {
       ? ''
       : '![CML watermark](https://raw.githubusercontent.com/iterative/cml/master/assets/watermark.svg)';
 
-    const report = `${userReport}\n\n${watermark}`;
+    let report = `${userReport}\n\n${watermark}`;
     const drv = getDriver(this);
 
+    const { remark } = await import('remark');
+    const { visit } = await import('unist-util-visit');
+
+    paths.add(markdownfile);
+
+    const publishLocalFiles = () => async (tree) => {
+      const nodes = [];
+
+      visit(tree, ['definition', 'image', 'link'], (node) => nodes.push(node));
+
+      const visitor = async (node) => {
+        if (node.url && node.alt !== 'CML watermark') {
+          try {
+            await fs.promises.access(node.url, fs.constants.F_OK);
+          } catch {
+            return;
+          }
+          node.url = await this.publish({ ...opts, path: node.url });
+          paths.add(node.url);
+        }
+      };
+
+      await Promise.all(nodes.map(visitor));
+    };
+
+    if (publish)
+      report = String(await remark().use(publishLocalFiles).process(report));
+
+    if (watch)
+      setInterval(() => {
```

Comment:
Since the CML watcher has to be called before the script that uses dvclive, could CML expose some var to indicate that this has been used, i.e. `CML_WATCH='markdownfile` ?

Reply:
'twould be great to have *all* CLI args default values to be overridable by environ (users may want to keep things secret) :)

Reply:
If you mean that running `cml ··· --watch` should set an environment variable in the parent process, no, that's technically impossible unless you resort to [non-portable methods](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).

Reply:
_Id est, the following is not possible:_

```console
$ printenv CML_WATCH
$ cml ··· --watch
$ printenv CML_WATCH # prints nothing, you expect the path to report.md
```

The easiest portable method is returning an `export` from the command:

```console
$ printenv CML_WATCH
$ $(cml ··· --watch)
$ printenv CML_WATCH # prints the path to report.md as expected
report.md
```

Reply:
I mean input vars:

```
run: cml pr
env:
  CML_some_opt: ...
```

Reply:
Maybe I misinterpreted @daavoo's “expose some var to indicate that this has been used” part of the comment. 🤔

Reply:
> Maybe I misinterpreted @daavoo's “expose some var to indicate that this has been used” part of the comment. 🤔

You didn't. I was asking for that.

> that's technically impossible unless you resort to [non-portable methods](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).

Could CML support and abstract away those non-portable methods, as it does for other features 👼  ? `cml set-env` ?

---

What I would like to have for DVCLive is an unambiguous signal that this feature has been used and, ideally, the path it's pointing to. Right now, I would need to use some proxy to check that 1. `we are in CI context` (I guess `CI` env var is ok`) and 2. `CML is installed`. I would be missing 3. `the user has actually called --watch` and 4. `the target path`.

It could also work if CML supports input vars, as per Casper's last comment, but this would rely on documenting the way we would like to set up the workflow but users could still end up facing some pitfalls

Reply:
For 3 & 4 `cml` could create a file on `/tmp` or something (?) 🪓

Reply:
> Right now, I could use some "proxy signals" and check that:
> - we are in CI context (I guess CI env var is ok)
> - CML is available.

I was envisioning something different.

1. DVCLive outputs markdown reports
2. CML watches for said reports

without any prior

0. DVCLive looks for CML

Reply:
Also known as:

```
cml whatever --watch report.md
python train.py report.md
```

Reply:
> Why should DVCLive need to know CML exists? 

Because markdown reports are not the default report format, `html` is. 

The idea of an extreme happy path is that the user won't need to change anything in the `train.py`  and DVCLive would decide whether or not to generate the `markdown` report.

It is not a deal-breaker, though. We can just rely on properly documenting the behavior and let users handle the report "type"

Reply:
> DVCLive would decide whether or not to generate [`md` or `html`] So, same code would generate [`html` (local) or `md` (CI)].

Hmm to me this sounds like a p2-nice-to-have or lower, but maybe

Reply:
Agree. it is minor and any of those would work.

Reply:
updated the OP/description & resolving here in favour of keeping discussion in #1036